code/software/memcheck/memcheck.c

```diff
@@ -392,8 +421,137 @@ int main(int argc, char **argv) {
     printf("\nComplete; Found a total of %ld bytes of writeable RAM\n\n", header->ram_total);
   }
 
-  printf("(Press a key)\n");
+  // perform memory tests until a key is pressed
+  char *memtest_start = (char *)((((uintptr_t)_end) + sizeof(MEMINFO) + 0xfffU) & ~0xfffU);
+  char *memtest_end   = memtest_start;
+
+  for (uint8_t i = 0; i < MAX_RAMBLOCKS; i++) {
+    if (header->blocks[i].block_size == 0) {
+      break;
+    }
+    // if onboard or expansion memory block
+    if ((header->blocks[i].flags & (RAMBLOCK_FLAG_ONBOARD | RAMBLOCK_FLAG_EXPANSION)) != 0) {
+      // block too low, skip it
+      if (header->blocks[i].block_start + header->blocks[i].block_size < (uintptr_t)memtest_start) {
+        continue;
+      }
+      // non-contiguous
+      if (header->blocks[i].block_start > (uintptr_t)memtest_end) {
+        break;
+      }
+      if (header->blocks[i].block_start + header->blocks[i].block_size >= (uintptr_t)memtest_end) {
+        memtest_end = (char *)header->blocks[i].block_start + header->blocks[i].block_size;
+      }
+    }
+  }
+
+  while (checkchar()) {  // clear any queued input
+    readchar();
+  }
+
+  printf("Continuous testing from 0x%06x to 0x%06x (press a key to exit)\n",
+         (unsigned int)memtest_start,
+         (unsigned int)memtest_end - 1);
+
+  int          memtest_words   = (memtest_end - memtest_start) / sizeof(uint16_t);
+  uint16_t    *word_ptr        = (uint16_t *)memtest_start;
+  int          update_interval = 0x7000;
+  int          pass            = 1;
+  int          badpasses       = 0;
+  int          goodpasses      = 0;
+  int          errors          = 0;
+  unsigned int start_ts        = _TIMER_100HZ;
+  for (;;) {
+    bool pass_bad = false;
+
+    init_LFSR();
+
+    for (int i = 0; i < memtest_words; i += update_interval) {
+      if (checkchar()) {
+        break;
+      }
+      unsigned int ts = _TIMER_100HZ - start_ts;
+      unsigned int tm = ts / (60 * 100);
+      ts              = (ts - (tm * (60 * 100))) / 100;
+
+      printf("\r%u:%02u %s Pass #%d - Filling words @ 0x%06x (LFSR 0x%04x)...    ",
+             tm,
+             ts,
+             errors ? "[BAD]" : "[OK]",
+             pass,
+             (unsigned int)&word_ptr[i],
+             (unsigned int)lfsr_val & 0xffff);
+      int k = memtest_words > (i + update_interval) ? (i + update_interval) : memtest_words;
+      for (int j = i; j < k; j++) {
+        uint16_t v  = next_LFSR();
+        word_ptr[j] = v;
+      }
+    }
+
+    if (checkchar()) {
+      break;
+    }
+
+#if 0  // fake error testing
```

Comment:
Just a question: do we want to keep this testing code?

Reply:
Well, maybe not, but I thought it was kind of useful for debugging to check error reporting "code path" so I left it in.

Reply:
If it concerns you I am happy to remove it. 🙂

Reply:
No probs, was just curious in case it had been missed :) Happy for it to stay (and can see the use for it).